### PR TITLE
[NEW] Added endpoint to retrieve mentions of a channel

### PR DESCRIPTION
**Pull request by MarcosSpessatto**
Added endpoint to retrieve all the mentions of a channel. Requested in [this](https://github.com/RocketChat/Rocket.Chat.iOS/issues/344#issuecomment-370359559) issue.

### CODE REVIEW

**Comment by MarcosSpessatto** [comment]
@rafaelks, could you take a look on this?

**Comment by MarcosSpessatto** [comment]
@rafaelks, exactly!

**Comment by rafaelks** [comment]
Added it to the milestone 👍

File: packages/rocketchat-mentions/server/methods/getUserMentionsByChannel.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+		const room = RocketChat.models.Rooms.findOneById(roomId);
+
+		if (!room) {
+			throw new Meteor.Error('error-invalid-room', 'Invalid room', { method: 'getUserMentionsByChannel' });
+		}
+
+		const user = RocketChat.models.Users.findOneById(Meteor.userId());
+
+		return RocketChat.models.Messages.findVisibleByMentionAndRoomId(user.username, roomId).fetch();
```
Comment (rodrigok) [inline packages/rocketchat-mentions/server/methods/getUserMentionsByChannel.js]: This should be paginated

File: packages/rocketchat-api/server/v1/channels.js [hunk 2]
```diff
@@ -784,3 +795,33 @@ RocketChat.API.v1.addRoute('channels.unarchive', { authRequired: true }, {
 		return RocketChat.API.v1.success();
 	}
 });
+
+RocketChat.API.v1.addRoute('channels.getAllUserMentionsByChannel', { authRequired: true }, {
+	get() {
+		const mountMentionObject = (mention) => {
+			return {
```
Comment (rodrigok) [inline packages/rocketchat-api/server/v1/channels.js]: Why are you changing the message format?

File: packages/rocketchat-api/server/v1/channels.js [hunk 3]
```diff
@@ -813,3 +813,30 @@ RocketChat.API.v1.addRoute('channels.unarchive', { authRequired: true }, {
...
+		const mentions = Meteor.runAsUser(this.userId, () => Meteor.call('getUserMentionsByChannel', {
+			roomId,
+			options: {
+				sort: sort ? sort : { ts: 1 },
+				skip: offset,
+				limit: count
+			}
+		}));
+
+		return RocketChat.API.v1.success({
```
Comment (rodrigok) [inline packages/rocketchat-api/server/v1/channels.js]: @MarcosSpessatto the other paginated APIs returns more data like:
```javascript
			return RocketChat.API.v1.success({
				channels: rooms,
				count: rooms.length,
				offset,
				total: RocketChat.models.Rooms.find(ourQuery).count()
			});
```

**Comment by rafaelks** [comment]
I just noticed that this API doesn't follow the way other APIs are working, with the prefix of `channels`, `dm` and `groups`. Do we want to change it, or keep it this way? Also, it's missing the docs. 👍

**Pull request merged**